docs/localization/LOCALIZATION.md

```diff
@@ -74,343 +78,443 @@ export function localizationConfigurationFactory() {
-import {SimpleHeaderPresComponent} from  './simple-header-pres.component';
-import {SimpleHeaderPresConfig} from  './simple-header-pres.config';
+#### Add localization to an existing component 
 
-@NgModule({
-  imports: [CommonModule, LocalizationModule],
-  declarations: [SimpleHeaderPresComponent],
-  exports: [SimpleHeaderPresComponent],
-  providers: [SimpleHeaderPresConfig]
-})
-export  class SimpleHeaderPresModule {}
+You can also localize a component later with:
+```shell
+ng g localization-to-component --path='path/to/the/component/class/file'
 ```
 
-- If you need to **only** translate **text** in your template file, simply use **translate pipe** or **translate directive** with/without parameters in the following way.
+As mentioned above, the `localization-to-component` schematic adds the localization architecture to a component. For an existing component, the schematic
+first checks that the component does not already contain localization files or properties, otherwise an error is thrown.
 
-```typescript
+> [!NOTE]
+> Later you can add a localization key to a component with the following command:
+> ```shell
+> ng g localization-key-to-component --path='path/to/the/component/class/file' --key='localizationKey' --value='Default value for localizationKey'
+> ```
+>
+> You can also do this using the Otter VSCode Extension that contains the `Otter: Add localization key to component` [feature](https://github.com/AmadeusITGroup/otter/blob/main/docs/vscode-extension/ADD_LOCALIZATION_KEY_TO_COMPONENT.md),
+> which largely simplifies this command.
 
-<!-- using a pipe -->
+### Generated files
 
-{{ "o3r-simple-header-pres.motto" | o3rTranslate }} // => this will output Let's shape the future of travel
-<!-- using a directive -->
+#### Localization file (`*.localization.json`)
 
-<div [translate]="o3r-simple-header-pres.locWithArg" [translateParams]="{user: 'otter friend'}"> // => this will output Hello, otter friend!
+The JSON localization file defines an object of key/value pairs. Each value is a JSON object with the properties `description` and `defaultValue`.
+Eventually, you can reference a global key via `$ref` using a relative path to a global localization JSON file or in a different package in the dependencies.
+The purpose of this file is to provide a default localization for a component so that the developer can start building pages using components without worrying about localization.
+The `*.localization.json` file specifies default values in English only.
 
+```json5
+{
+  "o3r-my-component-pres.somekey1": {
+    "description": "This is somekey1 description for translators",
+    "defaultValue": "This is my default value 1"
+  },
+  "o3r-my-component-pres.somekey2": {
+    "description": "This is somekey2 description for translators",
+    "defaultValue": "This is my default value 2"
+  },
+  "o3r-my-component-pres.someglobalkey1": {
+    "$ref": "../global.localization.json#/someglobalkey1"
```

Comment:
Maybe we are missing documentation to explain the purpose of using `$ref`

Reply:
There is a line above this code snippet to explain why we can use `$ref`, should we add additional documentation?